data_steward/tools/retract_data_gcs.py

```diff
@@ -124,25 +135,38 @@ def retract(pids, bucket, found_files, folder_prefix, force_flag):
             response = get_response()
         if response == "Y":
             # Output and input file content initialization
-            retracted_file_string = StringIO()
-            input_file_string = gcs_utils.get_object(bucket, folder_prefix + file_name)
-            input_file_lines = input_file_string.split('\n')
+            retracted_file_string = BytesIO()
+            input_file_bytes = gcs_utils.get_object(bucket, folder_prefix + file_name, as_text=False)
+            input_file_lines = input_file_bytes.split(b'\n')
             input_header = input_file_lines[0]
             input_contents = input_file_lines[1:]
-            retracted_file_string.write(input_header + '\n')
+            retracted_file_string.write(input_header + b'\n')
             logger.info("Checking for person_ids %s in path %s" % (pids, file_gcs_path))
 
             # Check if file has person_id in first or second column
             for input_line in input_contents:
-                if input_line != '':
-                    cols = input_line.split(",")
-                    col_1 = cols[0]
-                    col_2 = cols[1]
-                    if (table_name in PID_IN_COL1 and get_integer(col_1) in pids) or \
-                            (table_name in PID_IN_COL2 and get_integer(col_2) in pids):
-                        lines_removed += 1
+                # ensure line is not empty
+                if input_line.strip() != b'':
+                    input_line = input_line.strip()
```

Comment:
Why do the strip() twice?  Suggestion:
```
input_line = input_line.strip()
# ensure line is not empty
if input_line != b'':
    cols = input_lint.split(b',')
```

Reply:
I did this so that the original line is intact when writing back to the file if it is not retracted.

Reply:
I don't think writing back the same amount of white space is necessary on blank lines because line 151 makes it appear that we are expecting comma delimited fields.  But, if you don't like the above suggestion, you could also try:
```
if not input_line.isspace():
    input_line = input_line.strip()
    ...
```
This conditional ensures the `input_line` variable is never changed (even accidentally) and all the blank space can be rewritten as in the original.

Reply:
Yes, that works! Thanks